examples/chip-tool/BUILD.gn

```diff
@@ -34,8 +34,10 @@ executable("chip-tool") {
     "commands/discover/DiscoverCommand.cpp",
     "commands/discover/DiscoverCommissionablesCommand.cpp",
     "commands/discover/DiscoverCommissionersCommand.cpp",
-    "commands/pairing/CommissionedListCommand.cpp",
-    "commands/pairing/CommissionedListCommand.h",
+
+    # TODO - enable CommissionedListCommand once DNS Cache is implemented
```

Comment:
The cache will be for the DNS information of the operational nodes.

Before this PR we are storing the serialized CHIPDevice object in the persistent storage. The device object gets recreated when the app creates the device proxy for the peer device. It includes the PASESession, address and a lot of other information.

This was done as early on we were using PASE Session for secure session (before CASE was implemented). And, on reboot, the PASE based session was reconstructed from the storage.

Now we have support for CASE session, and it can be reestablished on reboots using local device credentials. So we don't really need to serialize and store the device object. Only information we need to persist is the device address, and that's the job of DNS cache.